sql/src/main/java/io/crate/analyze/symbol/Symbols.java

```diff
@@ -54,4 +57,51 @@ public DataType apply(@Nullable Symbol input) {
         }
         return streamers;
     }
+
+
+    /**
+     * returns true if the symbol contains the given columnIdent.
+     * If symbol is a Function the function tree will be traversed
+     */
+    public static boolean containsColumn(@Nullable Symbol symbol, ColumnIdent columnIdent) {
+        if (symbol == null) {
+            return false;
+        }
+        return HAS_COLUMN_VISITOR.process(symbol, columnIdent);
+    }
+
+    /**
+     * returns true if any of the symbols contains the given column
+     */
+    public static boolean containsColumn(Iterable<? extends Symbol> symbols, ColumnIdent columnIdent) {
+        for (Symbol symbol : symbols) {
+            if (containsColumn(symbol, columnIdent)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private static class HasColumnVisitor extends DefaultTraversalSymbolVisitor<ColumnIdent, Boolean> {
```

Comment:
can you use https://github.com/crate/crate/blob/9fd68b9790cf5a5fb4e48ecee0dbf5ba6a539f3b/sql/src/main/java/io/crate/analyze/symbol/SymbolVisitors.java here?